### PR TITLE
Build C++/CLI tests as .NET Core C++/CLI

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Update our C++/CLI tests to build with the `/clr:netcore` flag.

They'll normally build against the ref pack used to build the managed portions of the repo, but by passing `-cmakeargs -DCPP_CLI_LIVE_REF_ASSEMBLIES=1` on the command line for the `src/test/build.cmd` script, developers can build them against the in-repo ref pack.

I've validated locally that the `CPP_CLI_LIVE_REF_ASSEMBLIES` option repros the issues seen with static abstract interface members in WPF.

### CODE REVIEW

File: src/tests/Interop/IJW/getRefPackFolderFromArtifacts.ps1 [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+$testsFolder = Split-Path $interopFolder -Parent
+$srcFolder = Split-Path $testsFolder -Parent
+$repoRoot = Split-Path $srcFolder -Parent
+
+$versionPropsFile = "$repoRoot/eng/Versions.props"
+
+$majorVersion = Select-Xml -Path $versionPropsFile -XPath "/Project/PropertyGroup/MajorVersion" | %{$_.Node.InnerText}
+$minorVersion = Select-Xml -Path $versionPropsFile -XPath "/Project/PropertyGroup/MinorVersion" | %{$_.Node.InnerText}
+
+echo "$repoRoot/artifacts/bin/ref/net$majorVersion.$minorVersion"
```
Comment (AaronRobinsonMSFT) [inline src/tests/Interop/IJW/getRefPackFolderFromArtifacts.ps1:13]: Thoughts on writing out to stdout if this path doesn't exist? I am trying to ensure we know what failed if the CMake process fails. Same for the `getRefPackFolderFromSdk.ps1`

**Review by AaronRobinsonMSFT — APPROVED** [review]
We should add documentation for the way to build against live build. Perhaps in one of the test document markdown files or under the interop folder?

**Comment by davidwrighton** [comment]
I agree with @AaronRobinsonMSFT unless this is documented, this work will not be found or understood in the future when we need it. Please add something to our markdown.

**Comment by jkotas** [comment]
> CPP_CLI_LIVE_REF_ASSEMBLIES

Why can't we make this the default?

**Comment by jkoritzinsky** [comment]
We can't make this the default since in our current CI build, we build the native tests as part of the CoreCLR product build. So either we build the ref assemblies an extra time in the CoreCLR build which adds more time for every CI run, or we have to re-organize the build, which is a much higher cost than is budgeted for this PR.

**Review by davidwrighton — APPROVED** [review]

**Review by AaronRobinsonMSFT — APPROVED** [review]
The error handling is perfect. Much appreciated!

File: src/tests/Interop/IJW/getRefPackFolderFromArtifacts.ps1 [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+$versionPropsFile = "$repoRoot/eng/Versions.props"
+
+$majorVersion = Select-Xml -Path $versionPropsFile -XPath "/Project/PropertyGroup/MajorVersion" | %{$_.Node.InnerText}
+$minorVersion = Select-Xml -Path $versionPropsFile -XPath "/Project/PropertyGroup/MinorVersion" | %{$_.Node.InnerText}
+
+$refPackPath = "$repoRoot/artifacts/bin/ref/net$majorVersion.$minorVersion"
+
+if (-not (Test-Path $refPackPath))
+{
+    Write-Error "Reference assemblies not found in the artifacts folder. Did you build the libs.ref subset? Did the repo layout change?"
```
Comment (elinor-fung) [inline src/tests/Interop/IJW/getRefPackFolderFromArtifacts.ps1:17]: Include the expected path in the error message?

**Review by elinor-fung — APPROVED** [review]

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Pull request merged**